lua/ttt2/libraries/map.lua

```diff
@@ -504,6 +506,300 @@ function map.GetDataFromSpawnEntity(ent, spawnType)
     end
 end
 
+local mapsAll = {}
+local mapsPrefixes = {}
+local mapsWSIDs = {}
+
+if SERVER then
+    -- by default cs, de and test maps should be hidden
+
+    ---
+    -- @realm server
+    -- stylua: ignore
+    CreateConVar("ttt2_enable_map_prefix_cs", "0", { FCVAR_ARCHIVE, FCVAR_NOTIFY })
+
+    ---
+    -- @realm server
+    -- stylua: ignore
+    CreateConVar("ttt2_enable_map_prefix_de", "0", { FCVAR_ARCHIVE, FCVAR_NOTIFY })
+
+    ---
+    -- @realm server
+    -- stylua: ignore
+    CreateConVar("ttt2_enable_map_prefix_test", "0", { FCVAR_ARCHIVE, FCVAR_NOTIFY })
+
+    ---
+    -- Initializes the map list. Searches the file system for available maps, scans those maps
+    -- for their different prefixes and tries to associate a workshop ID with each map.
+    -- @internal
+    -- @realm server
+    function map.InitializeList()
+        local mapsFound = file.Find("maps/*.bsp", "GAME")
+
+        mapsAll = {}
```

Comment:
You are pulling ALL map files here and then you are painfully trying to match these to an addon below.
This is a lot of work and might even fail i assume.

Why not get all garrysmod maps and cstrike maps here.
Then below while iterating over the addons check if that addon contains a map file and you have the map title and the wsid without any matching.

eg.:
```lua
gmod_maps = file.Find("maps/*.bsp", "garrysmod")
cstrike_maps = file.Find("maps/*.bsp", "cstrike")
```

you can search each mounted addon by itself to see if it contains a map file:
```lua
addon_map = file.Find("maps/*.bsp", addon.title)
```

This would make this a lot cleaner IMO

Reply:
This is actually a good idea! only thing left: how do we handle the builtin maps (gmod, css etc) or locally installed maps then?

Reply:
> This is actually a good idea! only thing left: how do we handle the builtin maps (gmod, css etc) or locally installed maps then?

handle in what way exactly? we can find the maps via their respective searchpaths [1] with `file.Find` (see above).
Maybe we can get a list of mounted addons first so we don't miss any of those.

As for locally installed maps: if these are present as a .gma file they would still be accessible by the `addon.title`.
For maps in the `garrysmod/addons` folder i'm actually not sure.

We would probably need to iterate over all directories in the `garrysmod/addons` directory and then `file.Find` for each of those?

Pseudocode:
```
local _, directories = file.Find("addons/*", "garrysmod")

foreach directory in directories
	maps[directory] = file.Find("addons/" .. directory .. "/maps/*.bsp", "garrysmod")
```

not the nicest, but useable i think.

[1] https://wiki.facepunch.com/gmod/File_Search_Paths

Reply:
what abou the following:

1. create a map-name-indexed table of the workshop maps:
  ```
  {
    ["map_name_1"] = "687556776565786578",
    ["map_name_2"] = "76868768767",
    [....]
  }
  ```
2. iterater over all file system maps and assign the workshop ID by the previously generated lookup table
3. if there is no map found, use the old code from this PR to assign an icon

What's your opinion on this? This should be pretty fail safe while also supporting any map that is installed in any way

Reply:
i did not intend to have the icon assignment code changed (besides adapting to the proposed change),
but having a mapname -> wsid lookup table seems reasonable.

Reply:
Alright, that sounds good to me then. This should make the assignment pretty stable